pkg/user/manager/owncloudsql/owncloudsql.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/cs3org/reva/pkg/user"
+	"github.com/cs3org/reva/pkg/user/manager/owncloudsql/accounts"
+	"github.com/cs3org/reva/pkg/user/manager/registry"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+
+	// Provides mysql drivers
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func init() {
+	registry.Register("owncloudsql", NewMysql)
+}
+
+type manager struct {
+	c  *config
+	db *accounts.Accounts
+}
+
+type config struct {
+	DbUsername         string `mapstructure:"dbusername"`
+	DbPassword         string `mapstructure:"dbpassword"`
+	DbHost             string `mapstructure:"dbhost"`
+	DbPort             int    `mapstructure:"dbport"`
+	DbName             string `mapstructure:"dbname"`
+	Idp                string `mapstructure:"idp"`
+	Nobody             int64  `mapstructure:"nobody"`
+	JoinUsername       bool   `mapstructure:"join_username"`
+	JoinOwnCloudUUID   bool   `mapstructure:"join_ownclouduuid"`
+	EnableMedialSearch bool   `mapstructure:"enable_medial_search"`
+}
+
+// NewMysql returns a new user manager connection to an owncloud mysql database
+func NewMysql(m map[string]interface{}) (user.Manager, error) {
+	mgr := &manager{}
+	err := mgr.Configure(m)
+	if err != nil {
+		err = errors.Wrap(err, "error creating a new manager")
+		return nil, err
+	}
+
+	mgr.db, err = accounts.NewMysql(
+		fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", mgr.c.DbUsername, mgr.c.DbPassword, mgr.c.DbHost, mgr.c.DbPort, mgr.c.DbName),
+		mgr.c.JoinUsername,
+		mgr.c.JoinOwnCloudUUID,
+		mgr.c.EnableMedialSearch,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return mgr, nil
+}
+
+func (m *manager) Configure(ml map[string]interface{}) error {
+	c, err := parseConfig(ml)
+	if err != nil {
+		return err
+	}
+
+	if c.Nobody == 0 {
+		c.Nobody = 99
+	}
+
+	m.c = c
+	return nil
+}
+
+func parseConfig(m map[string]interface{}) (*config, error) {
+	c := &config{}
+	if err := mapstructure.Decode(m, &c); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func (m *manager) GetUser(ctx context.Context, uid *userpb.UserId) (*userpb.User, error) {
+	// search via the user_id
+	a, err := m.db.GetAccountByClaim(ctx, "userid", uid.OpaqueId)
+	if err == sql.ErrNoRows {
+		return nil, errtypes.NotFound(uid.OpaqueId)
+	}
+	return m.convertToCS3User(ctx, a)
+}
+
+func (m *manager) GetUserByClaim(ctx context.Context, claim, value string) (*userpb.User, error) {
+	a, err := m.db.GetAccountByClaim(ctx, claim, value)
+	if err == sql.ErrNoRows {
+		return nil, errtypes.NotFound(claim + "=" + value)
+	} else if err != nil {
+		return nil, err
+	}
+	return m.convertToCS3User(ctx, a)
+}
+
+func (m *manager) FindUsers(ctx context.Context, query string) ([]*userpb.User, error) {
+
+	accounts, err := m.db.FindAccounts(ctx, query)
+	if err == sql.ErrNoRows {
+		return nil, errtypes.NotFound("no users found for " + query)
+	} else if err != nil {
+		return nil, err
+	}
+
+	users := []*userpb.User{}
```

Comment:
This should be pre-allocated.
```suggestion
	users := make([]*userpb.User, 0, len(accounts))
```